text/0000-stabilize-arbitrary-self-types.md

```diff
@@ -32,25 +33,21 @@ When declaring a method, users can declare the type of the `self` receiver to be
 The `Receiver` trait is simple and only requires to specify the `Target` type to be resolved to:
 
 ```rust
-/// # SAFETY
-/// - the type needs to be ABI-compatible with a fat pointer
 trait Receiver {
     type Target: ?Sized;
 }
 ```
 
-
 The `Receiver` trait is already implemented for a few types from the standard, i.e.
-- smart pointer: `Arc<Self>`, `Box<Self>`, `Pin<Self>` and `Rc<Self>`
+- smart pointers: `Rc<Self>`, `Arc<Self>`, `Box<Self>`, and `Pin<Ptr<Self>>`, because these types all implement `Deref` and there's a blanket implementation of `Receiver` for `Deref`.
```

Comment:
Isn't `Pin<Self>` enough? `Pin<Ptr<Self>>` should be covered by recursiveness, if `Ptr` implements `Receiver`.

Reply:
I think `Pin<Self>` isn't actually right. See this error message: https://github.com/rust-lang/rust/blob/6f65ef57177ce0095171f70d0b010567c35e68cc/compiler/rustc_hir_analysis/src/check/wfcheck.rs#L1594

Reply:
True. Re-reading the pin docs it makes sense.